vendor/knative.dev/pkg/client/injection/kube/informers/factory/filtered/filtered_factory.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by injection-gen. DO NOT EDIT.
+
+package filteredFactory
+
+import (
+	context "context"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	informers "k8s.io/client-go/informers"
+	client "knative.dev/pkg/client/injection/kube/client"
+	controller "knative.dev/pkg/controller"
+	injection "knative.dev/pkg/injection"
+	logging "knative.dev/pkg/logging"
+)
+
+func init() {
+	injection.Default.RegisterInformerFactory(withInformerFactory)
+}
+
+// Key is used as the key for associating information with a context.Context.
+type Key struct {
+	Selector string
+}
+
+type LabelKey struct{}
+
+func WithSelectors(ctx context.Context, selector ...string) context.Context {
```

Comment:
seeing this in context Im wondering if we actually need this to take multiple `selector` strings or if a single `selector` would be enough. A selector can filter on multiple labels already with e.g "`label1,label1`" (or `labels.SelectorFromSet(..).String()`. Making this WithSelector might also make fixing the bug above easier.

Reply:
@julz ,  sorry for the late response.  I agree your point.  If no effort is done on this,  I would like to continue to fix the  problem in `pkg` module.